src/browser/transports/injection_bus.ts

```diff
@@ -88,28 +91,84 @@ export default class NativeWindowInjectionBus extends EventEmitter {
         return;
       }
 
-      // Ack / Nack
+      this._listenerHeartbeat(parsedMessage);
+
       if (this._pendingRequests.has(messageId)) {
-        const isNack = parsedMessage.action.includes('error');
-        const pendingRequest = this._pendingRequests.get(messageId);
+        this._listenerAck(parsedMessage);
+      } else {
+        this._listenerBroadcastMsg(parsedMessage);
+      }
+    };
 
-        isNack
-          ? pendingRequest.reject((<NackMessage>parsedMessage).payload.reason)
-          : pendingRequest.resolve();
+    copyDataTransport.on('message', this._messageListener);
+
+    this.subscribe();
+  }
 
-        this._senderId = senderId;
-        this._pendingRequests.delete(messageId);
+  // Setup heartbeat
+  private setupHeartbeat() {
+    this._connected = false;
+    this._previousDllSequence = -1;
```

Comment:
:neckbeard: Why not call this the same thing as what comes of the message (`const { sequence } = message;`)
:neckbeard: :neckbeard: `sequence` implies multiple, perhaps `sequenceNumber` is better

Reply:
1. There are two types of sequences: sequence number coming from the core, and sequence number coming from the message window. Wanted to be more explicit to possibly avoid confusion
2. Ok for `sequenceNumber `